src/device_impl.rs

```diff
@@ -178,46 +142,21 @@ where
         Ok(self.magnetometer_id()? == WHO_AM_I_M_VAL)
     }
 
-    /// Read temperature sensor data
-    pub fn temperature_data(&mut self) -> Result<i16, Error<CommE, PinE>> {
+    /// Get measured temperature.
+    pub fn temperature(&mut self) -> Result<Temperature, Error<CommE, PinE>> {
         let data = self
             .iface
             .read_accel_double_register(Register::OUT_TEMP_L_A)?;
-        Ok(data as i16)
-    }
 
-    /// Read temperature sensor data as celsius
-    pub fn temperature_celsius(&mut self) -> Result<f32, Error<CommE, PinE>> {
-        let data = self.temperature_data()?;
-        let temp_offset = (data as f32) / 256.0;
-        let default_temp = 25.0;
-        Ok(temp_offset + default_temp)
+        Ok(Temperature { raw: data })
     }
 
     /// Temperature sensor status
     pub fn temperature_status(&mut self) -> Result<TemperatureStatus, Error<CommE, PinE>> {
+        self.enable_temperature_sensor()?;
```

Comment:
Ah, I see. The enable method only sets it when necessary to avoid enabling it here every time.
While I understand the ease of use of enabling here if necessary, I would prefer to skip it and just do what the user asked. Otherwise, following the same logic, one would want to add a call to `init` in every method if not initialized.

What I would do is to document in this method that the temperature must be enabled beforehand.

Reply:
The reason I moved it here was because the temperature sensor was not working for me when enabled in `init`. It seems to only work when enabled after setting `AccelMode`.

Reply:
Reverted this change again.